exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -281,11 +282,29 @@ object SchemaUtils {
      *
      * @param databases the names of the databases
      * @param inBatch flag to perform database creation in a single batch
+     *
+     * For PostgreSQL, calls to this function should be preceded by connection.autoCommit = true,
+     * and followed by connection.autoCommit = false.
+     * @see org.jetbrains.exposed.sql.tests.shared.ddl.CreateDatabaseTest
      */
     fun createDatabase(vararg databases: String, inBatch: Boolean = false) {
-        with(TransactionManager.current()) {
-            val createStatements = databases.flatMap { listOf(currentDialect.createDatabase(it)) }
-            execStatements(inBatch, createStatements)
+        val transaction = TransactionManager.current()
+        try {
+            with(transaction) {
+                val createStatements = databases.flatMap { listOf(currentDialect.createDatabase(it)) }
+                execStatements(inBatch, createStatements)
+            }
+        } catch (e: ExposedSQLException) {
+            if (currentDialect.requiresAutoCommitOnCreateDrop && !transaction.connection.autoCommit) {
+                throw ExposedSQLException(
+                    cause = Exception(
+                        "${currentDialect.name} requires autoCommit to be enabled for CREATE DATABASE",
+                        e.cause
+                    ),
+                    e.contexts,
+                    transaction
+                )
+            } else throw e
```

Comment:
@bog-walk @e5l @Tapac @Tapac 
I'm not sure I like the way I did this. If there is a better way to do it, please let me know.

Reply:
We can try adding a constructor with a `message`, so it should be simple to change